thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -842,15 +845,27 @@ namespace diskann {
     const T *    query = data.scratch.aligned_query_T;
     const float *query_float = data.scratch.aligned_query_float;
 
-    for (uint32_t i = 0; i < this->data_dim; i++) {
+    auto q_dim = this->data_dim;
+    if(metric== diskann::Metric::INNER_PRODUCT && !this->use_sample){
+      // query_dim need to be specially treated when using IP
+      q_dim--;
+    }
+    for (uint32_t i = 0; i < q_dim; i++) {
       data.scratch.aligned_query_float[i] = query1[i];
       data.scratch.aligned_query_T[i] = query1[i];
       query_norm += query1[i] * query1[i];
     }
 
+
     // if inner product, we laso normalize the query and set the last coordinate
     // to 0 (this is the extra coordindate used to convert MIPS to L2 search)
     if (metric == diskann::Metric::INNER_PRODUCT) {
+      if(query_norm == 0){
+        // return an empty answer when calcu a zero point 
```

Comment:
Since query on 0 vector is illegal, do you think throw an exception is a better way to do?

Reply:
> Since query on 0 vector is illegal, do you think throw an exception is a better way to do?

No, in my opinion, zero point has its physical significance. see it 🗯️  [zero vector](https://qr.ae/pvuAh1). Returning an empty answer or a random answer list is both ok, but throwing an exception maybe not be a good idea, because zero vector can be used in mathematical calculations.
when it comes to a **NULL** vector, it should throw an exception.